enclave/src/lib.rs

```diff
@@ -425,6 +424,54 @@ pub unsafe extern "C" fn accept_pdex_accounts(
     sgx_status_t::SGX_SUCCESS
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn send_disk_data(encoded_data: *const u8, data_size: usize) -> sgx_status_t {
+    let mut data = slice::from_raw_parts(encoded_data, data_size);
+
+    let decoded: polkadex_sgx_primitives::StorageData =
+        if let Ok(data) = polkadex_sgx_primitives::StorageData::decode(&mut data) {
+            data
+        } else {
+            return sgx_status_t::SGX_ERROR_UNEXPECTED;
+        };
+
+    let balances_data = decoded.balances;
+    let nonce_data = decoded.nonce;
+    let orderbook_data = decoded.orderbook;
+
+    if polkadex_balance_storage::load_balance_storage().is_err()
+        && polkadex_balance_storage::create_in_memory_balance_storage().is_err()
```

Comment:
I'm not sure i understand this comment, the first check in the if verifies if the storage already exists, then for efficiency it uses the same if to return an error in case it doesn't already exist and creating it fails.

Reply:
Yes it does but because `polkadex_balance_storage::create_in_memory_balance_storage()`  does hard work it make sense to have it inside